02-technical-guides/aggregate-provider/main.ipynb

```diff
@@ -0,0 +1,1564 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "cb213d83-5c16-4b45-8967-abac205d0864",
+   "metadata": {},
+   "source": [
+    "# Understanding Aggregate Providers"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5b16c953-ab3b-4667-809a-498dbb043444",
+   "metadata": {},
+   "source": [
+    "An [aggregate provider](https://docs.atoti.io/latest/lib/atoti/atoti.aggregate_provider.aggregate_provider.html#atoti.AggregateProvider) precomputes and stores aggregates for the finest granularity of aggregation possible for certain hierarchies. This is also known as pre-aggregation. These intermediate aggregates are then stored in memory to speed up querying. \n",
+    "\n",
+    "In this notebook, we will look at how we can implement aggregate providers and how the performance varies with its addition.  \n",
+    "\n",
+    "<div style=\"text-align: center;\" ><a href=\"https://www.atoti.io/?utm_source=gallery&utm_content=aggregate-providers\" target=\"_blank\" rel=\"noopener noreferrer\"><img src=\"https://data.atoti.io/notebooks/banners/Discover-Atoti-now.png\" alt=\"Try atoti\"></a></div>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "2fefdac5-fe75-4c51-bfe7-e11a4f623c12",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import zipfile\n",
+    "\n",
+    "import wget\n",
+    "from IPython.display import clear_output, display\n",
+    "\n",
+    "import atoti as tt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "115f0050-162a-4922-8012-67517d3497b0",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def download_data(filename=\"small_data.zip\"):\n",
+    "    url = f\"https://data.atoti.io/notebooks/aggregate-provider/{filename}\"\n",
+    "    filename = wget.download(url)\n",
+    "\n",
+    "    with zipfile.ZipFile(filename, \"r\") as zip_ref:\n",
+    "        zip_ref.extractall()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "439adb00-b4dd-4993-90bb-f0000711b15d",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Downloading Data\n",
+    "\n",
+    "In order to test out the aggregate providers on server hosted on local machine, we prepared 2GB worth of data.  \n",
```

Comment:
I think this meant to say "In order to test out aggregate providers on a server hosted on a local machine, we prepared 2GB worth of data."